Sources/_StringProcessing/Regex/ASTConversion.swift

```diff
@@ -216,7 +216,7 @@ extension AST.Atom {
 
     switch self.kind {
     case let .char(c):                    return .char(c)
-    case let .scalar(s):                  return .char(Character(s.value))
+    case let .scalar(s):                  return .scalar(s.value)
```

Comment:
Do we want the behaviour of regex to be different depending on if we input a unicode value as its scalar code or as a literal character? ie `"ợ"` vs `"\u{1ee3}"`? With this one would be interpreted as a `.char` and the other as a `.scalar` in the DSLTree. I think the intent for the original code here was to make those two inputs equivalent

Reply:
I think they should be semantically equivalent (i.e they go through the exact same code paths in byte code emission), but this distinction allows us to have more accurate printing of the DSL tree when performing a literal -> DSL conversion

Reply:
If they behave the same then why would we need to print them differently? Printing them differently would imply that they behave differently I think.

Also yea, seeing as we're moving towards making a `.scalar` behave exactly the same as `.char`, maybe it would make sense to remove it outright

Reply:
> If they behave the same then why would we need to print them differently? Printing them differently would imply that they behave differently I think.

For example, if the user wrote `/e\u{301}/`, ideally the transformed DSL tree would be:

```swift
Regex {
  "e\u{301}"
}
```

instead of:

```swift
Regex {
  "é"
}
```

Semantically they're the same, but IMO it's much nicer to preserve the original syntax the user wrote.